### DISCUSSION TITLE
JS - Control Flow Statements - Decision Making Statements

### DISCUSSION THREAD

**Discussion by akash-coded** [akash-coded/mern#64] [Tasks]
King’s Island needs a program for its admission booths. When visitors to the park come up to the booth to
purchase their tickets, the worker uses this program to figure out how much to charge them. You will write
this program.

# Version 1
In the first version of the program, there is one ticket price of $30.00. Senior citizens (age ≥ 65) are given a
50% discount. Write this program as follows. Import your Scanner. Declare the needed variables (the person’s
age, the base price of a ticket ($30) and the price you will charge). Input the user’s age, compute the price of
the ticket and output the result in a formatted way (that is, using a $). You do not need to use DecimalFormat
for this part of the program but you will as you enhance it so you might want to set this up now. Save, compile
and run your program a few times, asking the user for different ages such as 10, 50, 65, 80 and 0.

# Version 2
The park wants to add further alterations to ticket costs. Children under 5 (less than 5, not less than or equal
to) are free. Ticket prices are now $40 unless the person is from Warren County in which case the tickets are
$30. Senior citizens still receive a 50% discount regardless of their county of residence. There are two input
parameters now, one for age and one for county. For the county, input this as a String. See below about how
to compare Strings. Calculate and output the ticket price. Ticket prices should be either 0, $15, $20, $30 or
$40 depending on the person’s age and location.

To compare Strings, we cannot use ==. Instead, we use a String message called equals. The format is
string.equals(string2). For instance, if your input variable is called county, then to see if county is Warren,
you would use county.equals(“Warren”). There is also equalsIgnoreCase if you do not want to
force the user to use proper capitalization. Save, compile and run your program using a variety of inputs to
make sure it works correctly.
We have two additional changes to make to the program. The first is that we want to ensure that the user’s age
is valid. A valid age is anything > 0. Use an if statement that will test to see if age <= 0 and if so, output
an error message that it is an invalid age. Otherwise, continue with the remainder of your code. You will have
to place the remainder of your code in a block.
Second, we want to give discounts to people from other counties as follows. Children under 14 from Clermont
County get an 18% discount (over the base price of $40) and senior citizens (>= 65) from Campbell County
get an additional 7.5% discount over their senior citizen discount. There are two ways you can do multiple
tests. The first is to use two nested if statements. For instance, if you want to test to see if someone is male
(‘M’) and age is greater than 20, you could do:
```
if(sex==‘M’)
    if(age>20)
```

Or, you can combine two tests in one conditional by ANDing the tests together. In Java, AND is indicated
using &&. The above code can be rewritten as if(sex==‘M’&&age>20). Use whichever approach you
prefer for the two new possibilities (Clermont children, Campbell seniors). Revise your program accordingly.
Compile, run and test your program. At this point, make sure you are formatting your output to be of the form
$xx.yy. Note: a 7.5% discount can be computed in two ways. You can do amount=amountamount*.075; (that is, subtract from amount 7.5% of amount) or amount=amount*.925; (make
amount be 92.5% of its current value). 

As this is a challenging set of logic, you might want to write down on paper all the conditions, how to determine
each one and how to compute the cost. Also, make sure you have commented your code well.

![image](https://user-images.githubusercontent.com/60097560/173000918-3dfcaf5e-06af-4774-9fa0-d8b9ba436f43.png)

**Comment by shubhamk-coder** [comment]
## Code 
### Version 1
```javascript
let age = parseInt(prompt("What is your age?"));
let basePrice = 30;
let price;
if (age >= 65) {
    price = basePrice - basePrice * .5;
    alert(`your ticket price is $ ${price}`);
} else {
    alert(`your ticket price is $ ${basePrice}`);
}
```
### Version 2
```javascript
let age = parseInt(prompt("What is your age?"));
if (age < 0){
    alert("It is an invalid age");
} else {
    let county = prompt("What is your county?");
    let basePrice = 40;
    let price;
    if (county === "Warren"){
        alert("your ticket price is $30");
    } else if (age < 5) {
        alert("Your ticket price is $0");
    } else if (age >= 65) {
        price = basePrice - basePrice * .5;
        alert(`your ticket price is $ ${price}`);
    } else if(age < 14 && county === "Clermont") {
        price = basePrice - basePrice * .18;
        alert(`your ticket price is $ ${price}`);
    } else if (age >= 65 && county === "Campbell") {
        price = basePrice - basePrice * .575;
        alert(`your ticket price is $ ${price}`);
    }
    else {
        alert(`your ticket price is $ ${basePrice}`);
    }
}
```

**Comment by Nikhi54** [comment]
### version 1

```javascript
let baseprice_ticket = 30.00;
let age = parseInt(prompt(`Enter your age !please:`));

let discount = .5;

let ticket_price = baseprice_ticket-baseprice_ticket*discount;

if(age>=65){
  console.log(`ticket price is $ ${ticket_price} `);
}
else{
  console.log(`ticket price is $ ${baseprice_ticket}`);
}
```

**Comment by Amrita-12345** [comment]
## Version1
```js
let baseprice=300;
let price;
for(i=1;i<=5;i++)
{
    let age=parseInt(prompt("Enter your age"));
if(age>=65)
{
    price=(300*(50/100));
    alert(`Price of ticket is ${price}`);
    }
else
{
    alert(`Price of ticket is ${baseprice}`);
}
}
```

## Version2
```js
let baseprice=500;
let baseprice1=700;
let price;
let age;
for(i=1;i<=12;i++)
{
    
    age=parseInt(prompt("Enter your age"));
    var country=prompt("Enter your country");
    // if(age==null)
    // {alert("Age is not entered")}
    if(age>=65 && country.match(/Warren/i))
   {
         price=((baseprice)*50/100);
         alert(`Price of ticket is ${price}`);
   }
   else if(age>=65 && country.match(/Campbell/i))
   {
         let price1=((baseprice1)*92.5/100);
          price=baseprice1-price1;
         alert(`Price of ticket is ${price}`);
   }
   else if(age>=65)
{
    price=(baseprice1*(50/100));
    alert(`Price of ticket is ${price}`);
    }
    else if(age<=0)
    {
        alert("Invalid age");
    }
    else if(age<5)
    {
        alert("Ticket is free");
    }
    else if(country.match(/Warren/i))
    {
        price=baseprice;
        alert(`price is ${price}`);
    }
    else if(age<14 && country.match(/Clermont/i))
   {
         price=((baseprice1)*18/100);
         alert(`Price of ticket is ${price}`);
   }
   
else
{
    alert(`Price of ticket is ${baseprice1}`);
}
}
```

**Comment by Neethumar** [comment]
### CODE

### Version 1

```javascript
let Age=parseInt(prompt("Enter your age"));
let cost=30;
var Totalcost=cost.toFixed(2);

if(Age>=65){
    Totalcost=(cost-cost*50/100).toFixed(2);
    alert(`Your ticket cost $${Totalcost}`);
}
else if(Age<65 && Age>0){
    alert(`Your ticket cost $${Totalcost}`);
}
else{
    alert("Please enter valid age")
}
```

**Comment by SumaGayathri-999** [comment]
### **CODE**

### version1
```javascript
let normal_price=30;
let age,price;
for(i=1;i<=5;i++)
{
    age=parseInt(prompt("enter your age:"));
    if(age>=65)
    {
        price=50*30/100;
        alert(`ticket price is $${price}`);
    }
    else
    {
        alert(`ticket price is $${normal_price}`);
    }
}

```
### version2
```javascript
let norm_price=40;
let price;
let age=parseInt(prompt("enter your age"));
let country=prompt("enter your country");
if(age<0)
{
   alert("invalid age")
}
else if(age<=5)
{
    alert("your ticket price is $0");
}
else if(age>=65)
{  
    let sr_price=norm_price-norm_price*(50/100);
    if(country==="warren")
    {
        let warren_price=30;
        price=warren_price-warren_price*(50/100);
        alert(`your ticket price is $${price}`);
    }
    else if(country==="campbell")
    {
        price=sr_price-sr_price*(7.5/100);
        alert(`your ticket price is $${price}`);
    }
    else
    {
    alert(`your ticket price is $${sr_price}`);
    }
}
else if(age<=14 && country==="clermont")
{
    price=norm_price-norm_price*(18/100);
    alert(`your ticket price is $${price}`)
}
else{
    alert(`your ticket price is $${norm_price}`)
}
```

**Comment by Ramgopalpatel** [comment]
Version1
```javascript
let baseprice=300;
let price;
for(i=1;i<=5;i++)
{
    let age=parseInt(prompt("Enter your age"));
if(age>=65)
{
    price=(300*(50/100));
    alert(`Price of ticket is ${price}`);
    }
else
{
    alert(`Price of ticket is ${baseprice}`);
}
}
```
Version2
```javascript
let age = parseInt(prompt("What is your age?"));
if (age < 0){
    alert("It is an invalid age");
} else {
    let county = prompt("What is your county?");
    let basePrice = 40;
    let price;
    if (county === "Warren"){
        alert("your ticket price is $30");
    } else if (age < 5) {
        alert("Your ticket price is $0");
    } else if (age >= 65) {
        price = basePrice - basePrice * .5;
        alert(`your ticket price is $ ${price}`);
    } else if(age < 14 && county === "Clermont") {
        price = basePrice - basePrice * .18;
        alert(`your ticket price is $ ${price}`);
    } else if (age >= 65 && county === "Campbell") {
        price = basePrice - basePrice * .575;
        alert(`your ticket price is $ ${price}`);
    }
    else {
        alert(`your ticket price is $ ${basePrice}`);
    }
}

```

**Comment by KNiranjani** [comment]
### Version1

**Code**
```js
let age=parseInt(prompt("Enter your age"));
let basePrice=30;let totalCost;

if(age>=65){
    totalCost=(basePrice-basePrice*50/100);
    console.log(`Your ticket cost $ ${totalCost}`);
}
else { 
    alert(`Your ticket cost $ ${basePrice}`);
}
```

**Comment by Thharshita** [comment]
> King’s Island needs a program for its admission booths. When visitors to the park come up to the booth to purchase their tickets, the worker uses this program to figure out how much to charge them. You will write this program.
> 
# Version 1
> In the first version of the program, there is one ticket price of $30.00. Senior citizens (age ≥ 65) are given a 50% discount. Write this program as follows. Import your Scanner. Declare the needed variables (the person’s age, the base price of a ticket ($30) and the price you will charge). Input the user’s age, compute the price of the ticket and output the result in a formatted way (that is, using a $). You do not need to use DecimalFormat for this part of the program but you will as you enhance it so you might want to set this up now. Save, compile and run your program a few times, asking the user for different ages such as 10, 50, 65, 80 and 0.


age= parseInt(prompt("Enter your age"))
price=30.
if (age>=65)
alert(`Pay amount  ${price*0.5}`)
else
alert(`pay amount ${price}`)


> 
 # Version 2
> The park wants to add further alterations to ticket costs. Children under 5 (less than 5, not less than or equal to) are free. Ticket prices are now $40 unless the person is from Warren County in which case the tickets are $30. Senior citizens still receive a 50% discount regardless of their county of residence. There are two input parameters now, one for age and one for county. For the county, input this as a String. See below about how to compare Strings. Calculate and output the ticket price. Ticket prices should be either 0, $15, $20, $30 or $40 depending on the person’s age and location.
> 




> To compare Strings, we cannot use ==. Instead, we use a String message called equals. The format is string.equals(string2). For instance, if your input variable is called county, then to see if county is Warren, you would use county.equals(“Warren”). There is also equalsIgnoreCase if you do not want to force the user to use proper capitalization. Save, compile and run your program using a variety of inputs to make sure it works correctly. We have two additional changes to make to the program. The first is that we want to ensure that the user’s age is valid. A valid age is anything > 0. Use an if statement that will test to see if age <= 0 and if so, output an error message that it is an invalid age. Otherwise, continue with the remainder of your code. You will have to place the remainder of your code in a block. Second, we want to give discounts to people from other counties as follows.
 Children under 14 from Clermont County get an 18% discount (over the base price of $40) and senior citizens (>= 65) from Campbell County get an additional 7.5% discount over their senior citizen discount. 


age = parseInt(prompt("Enter your age"))
county=prompt("Enter your county")
price=40
if(age<0)
alert("Invalid input")
else if (age<5)
alert(`Pay amount  $0`)
else if(age>65 && county.match(/Campbell/)))
alert(`Pay amount  ${price*0.50}`)
else if(age>65)
alert(`Pay amount  ${price*0.50}`)
else if(age>=5&& county.match(/Warren/))
alert(`pay amount $30`)
else if(age<14 && county.match(/Clermont County/))
alert(`Pay amount  ${price*0.18}`)

else 
alert(`pay amount ${price}`)

**Comment by sam2elect** [comment]
# CODE  VERSION-1 
```html
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha1/dist/css/bootstrap.min.css" rel="stylesheet"
        integrity="sha384-GLhlTQ8iRABdZLl6O3oVMWSktQOp6b7In1Zl3/Jr59b6EGGoI1aFkw7cmDA6j6gD" crossorigin="anonymous">
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha1/dist/js/bootstrap.bundle.min.js"
        integrity="sha384-w76AqPfDkMBDXo30jS1Sgez6pr3x5MlQ1ZAGC+nuZB+EYdgRZgiwxhTBTkF7CXvN"
        crossorigin="anonymous"></script>
    <script src="discussion64.js"></script>
    <style>
        @media only screen and (min-width: 600px) {
            .desktop{
                max-width: 40vw !important;
            }
        }

    </style>
    <title>Ticket Price Calculator</title>
</head>

<body>
    <section class="container  flex-column text-center pt-5">
        <section  class="gap-5 row justify-content-center d-flex flex-column align-items-center">

            <h1 class="h1 mb-4">Ticket Price Calculator</h1>

            <section class="input-group col-lg-6 desktop" >

                <label for="age" class="input-group-text ">Age:</label>

                <input type="number" id="age" class="form-control" required>
            </section>

            <button onclick="calculate()" class="btn btn-block btn-primary col-5 col-lg-2">Calculate</button>

            <p id="result" class="fw-bolder text-success"></p>

            <ul id="prices" class="fw-bolder text-success col-12 list-group col-lg-4">
            </ul>
        </section>
    </section>
</body>

</html>
```
```JS
function calculate() {
  var age = document.getElementById("age").value;
  const basePrice = 30.0;
  var ticketPrice;

  if (age >= 65) {
    ticketPrice = basePrice * 0.5;
  } else {
    ticketPrice = basePrice;
  }
  alert(`your ticket price is $ ${ticketPrice} for age $ ${age}`);
   
  document.getElementById("result").innerHTML =
   "<p>The price of the ticket is $" + ticketPrice + " for age : " + age</p>";

}
```

#SCREENSHORT
![image](https://user-images.githubusercontent.com/85131631/227764453-9655fb22-8c31-4879-be01-79b67333f421.png)


# CODE VERSION - 2

```html
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha1/dist/css/bootstrap.min.css" rel="stylesheet"
        integrity="sha384-GLhlTQ8iRABdZLl6O3oVMWSktQOp6b7In1Zl3/Jr59b6EGGoI1aFkw7cmDA6j6gD" crossorigin="anonymous">
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha1/dist/js/bootstrap.bundle.min.js"
        integrity="sha384-w76AqPfDkMBDXo30jS1Sgez6pr3x5MlQ1ZAGC+nuZB+EYdgRZgiwxhTBTkF7CXvN"
        crossorigin="anonymous"></script>
    <script src="discussion64.js"></script>
    <style>
        @media only screen and (min-width: 600px) {
            .desktop{
                max-width: 40vw !important;
            }
        }

    </style>
    <title>Ticket Price Calculator</title>
</head>

<body>
    <section class="container  flex-column text-center pt-5">
        <section  class="gap-5 row justify-content-center d-flex flex-column align-items-center">

            <h1 class="h1 mb-4">Ticket Price Calculator</h1>

            <section class="input-group col-lg-6 desktop" >

                <label for="age" class="input-group-text ">Age:</label>

                <input type="number" id="age" class="form-control" required>
            </section>

            <section class="input-group col-lg-6 desktop" >

                <label for="country" class="input-group-text ">Country:</label>

                <input type="text" id="country" class="form-control" required>
            </section>

            <button onclick="calculate()" class="btn btn-block btn-primary col-5 col-lg-2">Calculate</button>

            <p id="result" class="fw-bolder text-success"></p>

            <ul id="priceList" class="fw-bolder text-success col-12 list-group col-lg-4">
            </ul>
        </section>
    </section>
</body>

</html>
```
```js

function calculate() {
  var age = document.getElementById("age").value;

  let country = document.getElementById("country").value.toLowerCase();

  var basePrice = 40;
    var price = 0;
  var ticketPrice;

  if (age <= 0) {
    alert("Invalid age!");
    return;
  }

  //calc the base price acording to the country
  if (country == "warren") {
    basePrice = 30;
  }

  //apply age disc
  if (age < 5) {
    price = 0;
  } 
  
  else if (age >= 5 && age < 14 && country == "clermont") {
    price = basePrice * 0.82;
  } 
  
  else if (age >= 65) {
    price = basePrice * 0.5;

    if (country == "campbell") {
      price *= 0.925;
    }
  } 
  
  else {
    price = basePrice;
  }

  alert(`your ticket price is $ ${price} for age $ ${age} where country is $ ${country}`);

    document.getElementById("result").innerHTML =
    "<p>The price of the ticket is $" + price + " for age :  " + age + "  where country is :  " + country</p>";

}

```
# SCREENSHORT
![image](https://user-images.githubusercontent.com/85131631/227766972-e8f704fd-c0b9-4202-94fb-43c3cb511969.png)

**Comment by kumarneeraj89** [comment]
# Code

```javascript
let ticket_base_price = 40;
let discount_senior_citizens = 50;
let persons_age = 64;
let county = 'india';


if (persons_age>=60){
    if(persons_age>=65){
          if (county === 'clermont' || county === 'Clermont') {
               ticket_base_price =  (ticket_base_price - ((ticket_base_price*18)/100) - ((ticket_base_price*7.5)/100) );
                console.log("Ticket Price: $" + ticket_base_price);
        }
        else{
            ticket_base_price = ticket_base_price/2;
        console.log("Ticket Price: $" + ticket_base_price);
        }
    }
    else if (county === 'Warren' || county === 'warren'){
            ticket_base_price = 30/2;
            console.log("Ticket Price: $" + ticket_base_price);
        }
        
    else{
        ticket_base_price = ticket_base_price/2;
        console.log("Ticket Price: $" + ticket_base_price);
    }
}
else if (persons_age>14 && persons_age<=60){
    if(county === 'Warren' || county === 'warren'){
        ticket_base_price = 30;
        console.log("Ticket Price: $" + ticket_base_price);
    }
    else{
    console.log("Ticket Price: $" + ticket_base_price);
    }
}

else if (persons_age<0){  
    console.log("Invalid Age");
}

else if (persons_age>=0 && persons_age<=14){
      if(persons_age > 5){
        if(county === 'Clermont' || county === 'clermont'){
             console.log("Ticket Price: $" + (ticket_base_price - (ticket_base_price*18)/100));
        }
        else{
            console.log("Ticket Price: $" + ticket_base_price);
        }
      }
      else{
        console.log("Ticket Price: $0");
      }
}

else{
    console.log("Ticket Price: $" + ticket_base_price);
}

**Comment by vk0027** [comment]
Version 1

let baseprice_ticket = 30.00;
let age = parseInt(prompt(Enter your age !please:));

let discount = 0.5;

let ticket_price = baseprice_ticket-baseprice_ticket*discount;

if(age>=65){
console.log(ticket price is $ ${ticket_price} );
}
else{
console.log(ticket price is $ ${baseprice_ticket});
}

Version 2

let ticket_base_price = 40;
let discount_senior_citizens = 50;
let persons_age = 6;
let county = 'clermont';


if (persons_age>=60){
    if(persons_age>=65){
          if (county === 'clermont' || county === 'Clermont') {
               ticket_base_price =  (ticket_base_price - ((ticket_base_price*18)/100) - ((ticket_base_price*7.5)/100) );
                console.log("Ticket Price: $" + ticket_base_price);
        }
    }
        else if (county === 'Warren' || county === 'warren'){
            ticket_base_price = 30/2;
            console.log("Ticket Price: $" + ticket_base_price);
        }
        
         else{
            ticket_base_price = ticket_base_price/2;
            console.log("Ticket Price: $" + ticket_base_price);
        }
}
else if (persons_age>14 && persons_age<=60){
    if(county === 'Warren' || county === 'warren'){
        ticket_base_price = 30;
        console.log("Ticket Price: $" + ticket_base_price);
    }
    else{
    console.log("Ticket Price: $" + ticket_base_price);
    }
}

else if (persons_age<0){  
    console.log("Invalid Age");
}

else if (persons_age>=0 && persons_age<=14){
      if(persons_age > 5){
        if(county === 'Clermont' || county === 'clermont'){
             console.log("Ticket Price: $" + (ticket_base_price - (ticket_base_price*18)/100));
        }
        else{
            console.log("Ticket Price: $" + ticket_base_price);
        }
      }
      else{
        console.log("Ticket Price: $0");
      }
}

else{
    console.log("Ticket Price: $" + ticket_base_price);
}

**Comment by RevathiSundaramoorthy** [comment]
## Code
### Version 1
```javascript
let basePrice = 30;
let age = parseInt(prompt("Enter the visitor's age"));
if (age <= 0) {
	console.log ("Please enter a valid age");
}
else {
	let ticketPrice = basePrice;
//50% discount for senior citizens
	if (age >= 65) {
  	  ticketPrice = basePrice * 0.5;
	}
console.log ("Total Ticket price to be paid is $" + ticketPrice);
}
```
### Version 2
```javascript
let basePrice = 40;
let basePrice_Warren = 30;
let ticketPrice = 0.00;
let age = parseInt(prompt("Enter the visitor's age"));
let county = prompt("Enter the visitor's county of residence");
if (age <= 0) {
    console.log ("Please enter a valid age");
}
else{ 
    if (age >= 5) {
        if (county.toUpperCase() == 'WARREN') {
            ticketPrice = basePrice_Warren;
        }
        else {
	        ticketPrice = basePrice;
        }
        //50% discount for senior citizens
	    if (age >= 65) {
  	        ticketPrice = ticketPrice * 0.5;
	    }
    }
    if ((age < 14) && (county.toUpperCase() == 'CLERMONT')) {
        ticketPrice = ticketPrice - (ticketPrice * 0.18);
    }
    if ((age >= 65) && (county.toUpperCase() == 'CAMPBELL')) {
        ticketPrice = ticketPrice - (ticketPrice * 0.075);
    }
    console.log ("Visitor's age: " + age);
    console.log ("Visitor's county of residence: " + county);
    console.log ("Total Ticket price to be paid is $" + ticketPrice);
}
```
## Screenshot
![image](https://user-images.githubusercontent.com/126975777/233802839-b5f99d34-bb88-4e90-b76a-a7e580129e28.png)
![image](https://user-images.githubusercontent.com/126975777/233802860-e187edd3-7291-4a19-88d5-9352fc2e3c47.png)
![image](https://user-images.githubusercontent.com/126975777/233802875-c450676c-a2af-414f-85b7-42dbe89bc01d.png)
![image](https://user-images.githubusercontent.com/126975777/233802937-ff9f1b4c-d153-451c-9312-a7a461738e31.png)
![image](https://user-images.githubusercontent.com/126975777/233802950-2a0b0d25-4107-42e5-ad9b-04f838488c6d.png)
